src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -1245,23 +1245,39 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
-                const auto well_status = Well::StatusFromString( status_str );
+            if (conn_defaulted(record)) {
+                const auto new_well_status = Well::StatusFromString(status_str);
+
                 for (const auto& wname : well_names) {
-                    {
-                        const auto& well = this->getWell(wname, currentStep);
-                        if( well_status == open && !well.canOpen() ) {
-                            auto elapsed = this->snapshots[currentStep].start_time() - this->snapshots[0].start_time();
-                            auto days = std::chrono::duration_cast<std::chrono::hours>(elapsed).count() / 24;
-                            std::string msg = "Well " + wname
-                                + " where crossflow is banned has zero total rate."
-                                + " This well is prevented from opening at "
-                                + std::to_string( days ) + " days";
-                            OpmLog::note(msg);
-                        } else {
-                            this->updateWellStatus( wname, currentStep, well_status);
-                            if (handlerContext.sim_update)
-                                handlerContext.sim_update->affected_wells.insert(wname);
+                    if ((new_well_status == open) && !this->getWell(wname, currentStep).canOpen()) {
+                        auto elapsed = this->snapshots[currentStep].start_time() - this->snapshots[0].start_time();
+                        auto days = std::chrono::duration_cast<std::chrono::hours>(elapsed).count() / 24;
+                        std::string msg = "Well " + wname
+                            + " where crossflow is banned has zero total rate."
+                            + " This well is prevented from opening at "
+                            + std::to_string( days ) + " days";
+                        OpmLog::note(msg);
+                    }
+                    else {
+                        const auto did_update_well_status =
+                            this->updateWellStatus(wname, currentStep, new_well_status);
+
+                        if (handlerContext.sim_update) {
+                            handlerContext.sim_update->affected_wells.insert(wname);
+                        }
+
+                        if (did_update_well_status && (new_well_status == open)) {
+                            // Record possible well injection/production status change
+                            auto well2 = this->snapshots[currentStep].wells.get(wname);
+
+                            const auto did_flow_update =
+                                (well2.isProducer() && well2.updateHasProduced())
+                                ||
+                                (well2.isInjector() && well2.updateHasInjected());
+
+                            if (did_flow_update) {
+                                this->snapshots[currentStep].wells.update(std::move(well2));
```

Comment:
Is this your rule? Always use move whenever you pass a value?

What does the compiler do here (function takes parameter by value)?

Reply:
> Is this your rule? Always use move whenever you pass a value?

I try to write `move(x)` to signal that I'm transferring ownership of the `x` to the container/callee.  In this case [`map_member<>::update()`](https://github.com/OPM/opm-common/blob/31cddd2d866760cebbec69205e955a841ee8e372/opm/input/eclipse/Schedule/ScheduleState.hpp#L184-L187) takes its parameter by value and I have a value of the appropriate type that I'm not going to reference after the call to `update()`.